cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -189,6 +192,13 @@ func (p *dockerProject) Build(
 ) *async.TaskWithProgress[*ServiceBuildResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServiceBuildResult, ServiceProgress]) {
+			if serviceConfig.Docker.RemoteBuild {
```

Comment:
This feels like a place where our existing abstractions break down. I'm not sure what build should do. In the local build model, we do the following:

1. Build: Run `docker build` but don't assign a local build tag or remote build tag.
2. Package: assign a local tag to the image
3. Deploy: Push the local image to the remote registry.

With remote build, we must couple building the image with pushing it to the remote registry, so we can't use this same split.

We could consider having `build` be a no-op during this mode but have `package` build the docker build context `.tar.gz`.  That might be useful as that could be a reasonable thing to provide `--from-package` during Deploy?

Reply:
Yeah, it feels like like build/package is not supported when using remote build and that it needs to be handled in Deploy.